### PR TITLE
05875 query workflow improvements part2

### PR DESCRIPTION
**Pull request by netopyr**
Another cleanup run of the query workflow

- Introduced `QueryContext`
- Adjusted all `QueryHandlers`
- Added hedera-app interfaces for `SolvencyCheck`
- All facilities used during query also throw Exceptions now
- Reused ingest checks for CryptoTransfer within query
- ...

Still missing, will be done in follow-up PR:
- Cleanup of fee calculation
- Cleanup of account balance check

### CODE REVIEW

File: hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/workflows/ExceptionConditions.java
Comment (rbair23) [inline hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/workflows/ExceptionConditions.java:1]: Interesting. I added https://github.com/hashgraph/hedera-services/pull/6063/files/360ddc9a1b0d1e50fab27ce052d6ad4474f98baa#diff-0c7531a627dbf0b47150215c83e6428bf24b7c68c54bf64877669f398c6b7386in my PR that helps with doing hedera-specific assertions for tests. Maybe the two approaches can be combined?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/solvency/SolvencyPreCheck.java
Comment (rbair23) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/solvency/SolvencyPreCheck.java:1]: Does this require a whole interface or should the solvency logic be moved to ... somewhere in `hedera-app`? What inputs does the solvency check require?

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryCheckerTest.java
Comment (rbair23) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryCheckerTest.java:229]: This is super cool!

**Review by rbair23 — APPROVED** [review]

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryCheckerTest.java
Comment (netopyr) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryCheckerTest.java:229]: Thanks!

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/solvency/SolvencyPreCheck.java
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/solvency/SolvencyPreCheck.java:1]: Good question. For now I tried to stick as close as possible to the structure in mono-services to keep it simple. But once our workflows are working and we had some time to take a closer look at the implementation of the facilities, we should discuss if we want to restructure them.

File: hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/workflows/ExceptionConditions.java
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/workflows/ExceptionConditions.java:1]: I saw that, but I was not sure yet, what to think about it. It clearly simplifies the code, but it also kind of breaks with the principles of AspectJ, at least in my opinion.

**Review by jeromy-cannon — APPROVED** [review]
from a release engineering perspective I approve the build changes, please wait for other approvers for remaining code before merging.

**Review by rbair23 — APPROVED** [review]

**Comment by jsync-swirlds** [comment]
Change to ScheduleService looks good.

**Review by iwsimon — APPROVED** [review]
LGTM

**Pull request merged**
